pkgs/tools/nix/nix-monitored/default.nix

```
@@ -0,0 +1,51 @@
+{ lib
+, fetchFromGitHub
+, stdenv
+, nix
+, nix-output-monitor
+}:
+
+stdenv.mkDerivation {
+  inherit (nix)
+    outputs
+    version
+    ;
+  pname = "nix-monitored";
+  src = fetchFromGitHub {
+    owner = "ners";
+    repo = "nix-monitored";
+    rev = "29a7741768aabb1a9934f2dba6bd50918c78b33b";
+    hash = "sha256-LhRefSAtUrEPkc/4P7BEU2jeJQ4tD+wp+UKTe0N1EF0=";
+  };
+  makeFlags = [ "BIN=nix" "BINDIR=$(out)/bin" "NIXPATH=${lib.makeBinPath [ nix nix-output-monitor ]}" ];
+  postInstall = ''
+    ln -s $out/bin/nix $out/bin/nix-build
+    ln -s $out/bin/nix $out/bin/nix-shell
+    ls ${nix} | while read d; do
+      [ -e "$out/$d" ] || ln -s ${nix}/$d $out/$d
+    done
+    ls ${nix}/bin | while read b; do
+      [ -e $out/bin/$b ] || ln -s ${nix}/bin/$b $out/bin/$b
+    done
```

shellcheck will complain about. The better way is to do the following:
Also we don't need to check if the file exist if we iterate over all existing files.

```suggestion
    for file in ${nix}; do
      ln -s ${nix}/$file $out/$file
    done
    for file in ${nix}/bin; do
      ln -s ${nix}/bin/$file $out/bin/$file
    done
```
Also this sounds like an ideal usecase for xorg.lndir which creates nested symlink structures.

---

`for file in <directory>` doesn't actually go through the contents of the directory, it just iterates the single element, namely `<directory>`.
I guess you meant `for file in <directory>/*`.

The reason for the file existing check is that some files, namely `nix`, `nix-build`, and `nix-shell`, are linked separately, and this loop allows us to easily skip over them.

---

The use-case of this script is:
 - link everything from `${nix}` except for `/bin/nix`, `/bin/nix-build`, `/bin/nix-shell`
 - so we can link every directory other than `/bin`
 - we can link every file in `/bin` except for those three

I don't think a shorter or more straightforward script can be written than this, even with `lndir`.